components/core/DataView.js

```diff
@@ -230,56 +197,55 @@ export default class DataView extends React.Component {
     startIndex: 0,
     checked: {},
     view: "grid",
+    viewLimit: 20,
   };
 
   async componentDidMount() {
     if (!mounted) {
       mounted = true;
       window.addEventListener("remote-data-deletion", this._handleDataDeletion);
-      window.addEventListener(
-        "remote-slate-object-remove",
-        this._handleRemoteSlateObjectRemove
-      );
-      window.addEventListener(
-        "remote-slate-object-add",
-        this._handleRemoteSlateObjectAdd
-      );
+      window.addEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
+      window.addEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
       window.addEventListener("remote-update-carousel", this._handleUpdate);
     }
 
+    window.addEventListener("scroll", this._handleScroll);
+
     await this._handleUpdate();
   }
 
   componentWillUnmount() {
     mounted = false;
 
-    window.removeEventListener(
-      "remote-data-deletion",
-      this._handleDataDeletion
-    );
-    window.removeEventListener(
-      "remote-slate-object-remove",
-      this._handleRemoteSlateObjectRemove
-    );
-    window.removeEventListener(
-      "remote-slate-object-add",
-      this._handleRemoteSlateObjectAdd
-    );
+    window.removeEventListener("remote-data-deletion", this._handleDataDeletion);
+    window.removeEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
+    window.removeEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
+
+    window.removeEventListener("scroll", this._handleScroll);
     window.removeEventListener("remote-update-carousel", this._handleUpdate);
   }
 
-  _increment = (direction) => {
-    if (
-      direction > 0 &&
-      this.state.startIndex + VIEW_LIMIT <
-        this.props.viewer.library[0].children.length
-    ) {
-      this.setState({ startIndex: this.state.startIndex + VIEW_LIMIT });
-    } else if (direction < 0 && this.state.startIndex - VIEW_LIMIT >= 0) {
-      this.setState({ startIndex: this.state.startIndex - VIEW_LIMIT });
+  _handleScroll = (e) => {
+    const windowHeight =
+      "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;
+    const body = document.body;
+    const html = document.documentElement;
+    const docHeight = Math.max(
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const windowBottom = windowHeight + window.pageYOffset;
+    if (windowBottom >= docHeight) {
+      this.setState({ viewLimit: this.state.viewLimit + 20 });
+    } else {
```

Comment:
✂️ cut this empty else?

Also might be useful to look at `common/window` and make these some utility methods we can use again.

Reply:
You might want to add a debounce to this if you're listening to scroll, so it doesn't trigger setState an insane amount of times.